library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -99,10 +101,9 @@ int getBatchStatus(long batchId) {
         }
 
         boolean hasCompleteItems = statusCountMap.hasCountFor(DownloadStatus.SUCCESS);
-        boolean hasPendingItems = statusCountMap.hasCountFor(DownloadStatus.PENDING);
         boolean hasSubmittedItems = statusCountMap.hasCountFor(DownloadStatus.SUBMITTED);
         boolean hasOtherItems = statusCountMap.hasNoItemsWithStatuses(STATUSES_EXCEPT_PENDING_SUCCESS_SUBMITTED);
-        if (hasCompleteItems && (hasPendingItems || hasSubmittedItems) && !hasOtherItems) {
+        if (hasCompleteItems && hasSubmittedItems && !hasOtherItems) {
```

Comment:
Pending tasks don't imply that the batch is running. This causes the following bug:
- There are `SUCCESS` and `SUBMITTED` and `PENDING` tasks, the batch is running
- The submitted task finishes and moves to success, the `DownloadService` dies for any reason, like a phone reboot
- The application restarts again, the DownloadService starts again, reads the status of the batch and as there are success and pending tasks, the batch is active
- When the batch is active, no download is moved to the submitted status and no download is added to the executor.
- The DownloadStatus stays active forever and no tasks going to be added to the DownloadThread, nothing is downloading and all pending tasks stay pending.